From 21bd91156255b55dff9e8e68a6e4b4d1bf87b2ca Mon Sep 17 00:00:00 2001
From: Mingliang Liu <liuml07@apache.org>
Date: Thu, 22 Dec 2016 17:09:26 -0800
Subject: [PATCH 2234/2848] HADOOP-13937. S3Guard: Mock bucket locations in
 MockS3ClientFactory. Contributed by Mingliang Liu

(cherry picked from commit b273171896c51c1f7f4680637e83118de385e754)

Change-Id: I0b222dadfdc287e7c9737b36ed89eb8e8bfa8a9c
---
 .../apache/hadoop/fs/s3a/MockS3ClientFactory.java  |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/MockS3ClientFactory.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/MockS3ClientFactory.java
index 41f04ee..fe04b9a 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/MockS3ClientFactory.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/MockS3ClientFactory.java
@@ -23,6 +23,7 @@
 import java.net.URI;
 
 import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.Region;
 
 /**
  * An {@link S3ClientFactory} that returns Mockito mocks of the {@link AmazonS3}
@@ -35,6 +36,8 @@ public AmazonS3 createS3Client(URI name, URI uri) {
     String bucket = name.getHost();
     AmazonS3 s3 = mock(AmazonS3.class);
     when(s3.doesBucketExist(bucket)).thenReturn(true);
+    when(s3.getBucketLocation(anyString()))
+        .thenReturn(Region.US_West.toString());
     return s3;
   }
 }
-- 
1.7.9.5

